src/devices/Mcp23xxx/Mcp23008.cs

```diff
@@ -48,23 +61,75 @@ public byte[] Read(Register.Address startingRegisterAddress, byte byteCount)
         }
 
         /// <summary>
-        ///  Writes a byte to a register.
+        /// Sets a mode to a pin.
+        /// </summary>
+        /// <param name="pinNumber">The pin number.</param>
+        /// <param name="mode">The mode to be set.</param>
+        public void SetPinMode(int pinNumber, PinMode mode)
+        {
+            ValidateMode(mode);
+            ValidatePin(pinNumber);
+            _ioDirection = ChangePin(
+                Register.Address.IODIR,
+                pinNumber,
+                mode == PinMode.Input ? PinValue.High : PinValue.Low,
+                Read(Register.Address.IODIR));
+        }
+
+        /// <summary>
+        /// Reads the value of a pin.
         /// </summary>
-        /// <param name="registerAddress">The register address to write.</param>
-        /// <param name="data">The data to write to the register.</param>
-        public void Write(Register.Address registerAddress, byte data)
+        /// <param name="pinNumber">The pin number.</param>
+        /// <returns>High or low pin value.</returns>
+        public PinValue ReadPin(int pinNumber)
         {
-            Write(registerAddress, data, Port.PortA, Bank.Bank1);
+            ValidatePin(pinNumber);
+
+            byte data = Read(Register.Address.GPIO);
+            return ((data & (1 << (pinNumber % 8))) > 0) ? PinValue.High : PinValue.Low;
         }
 
         /// <summary>
-        /// Writes a number of bytes to registers.
+        /// Writes a value to a pin.
         /// </summary>
-        /// <param name="startingRegisterAddress">The starting register address to write.</param>
-        /// <param name="data">The data to write to registers.</param>
-        public void Write(Register.Address startingRegisterAddress, byte[] data)
+        /// <param name="pinNumber">The pin number.</param>
+        /// <param name="value">The value to be written.</param>
+        public void WritePin(int pinNumber, PinValue value)
+        {
+            ValidatePin(pinNumber);
+            ChangePin(Register.Address.GPIO, pinNumber, value, Read(Register.Address.OLAT));
```

Comment:
Two comments here:
1.- Seems to me you have a bug here and are reading OLAT instead of GPIO.
2.- Should we cache gpio values similar to how you cache _ioDirection in order to avoid two operations?

Reply:
You read the current value from OLAT.  Writing GPIO changes OLAT.

Reply:
And I'm not caching the other- neither does adafruit. I think we could consider it as an option and I'll call it out with the direction validation as an improvement followup.

Reply:
Adafruit does cache, at least on their python code: https://github.com/adafruit/Adafruit_CircuitPython_MCP230xx/blob/f3d9cbefb4f440a96d2b86c725bda7b67993344e/adafruit_mcp230xx.py#L268-L270

Reply:
Whoops. I was looking at legacy Adafruit_MCP230xx. Yeah, we should follow suit. But do we have to block this PR for it? If so I'll be making a massive change as I don't want to write that on the unrefactored code (which I have in progress).

Reply:
Er, never mind. They don't cache in this code either. Not terribly used to reading Python. It's just calling the property to set the new value for those registers.